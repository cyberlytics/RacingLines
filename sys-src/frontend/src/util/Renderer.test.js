require('jest-canvas-mock');

const { Player } = require('./Player');
const {Renderer} = require('./Renderer');


//Test if object has all functions
describe("Test if renderer has all functions of the class Renderer", () => {
  const renderer = new Renderer();

  test("Test if drawPlayers is defined", () => {
    expect(typeof renderer.drawPlayers).toBe("function");
  })

  test("Test if drawLines is defined", () => {
    expect(typeof renderer.drawLines).toBe("function");
  })

})


//The following tests got written with the help of the article
//https://yonatankra.com/how-to-test-html5-canvas-with-jest/
//Test function drawPlayers()
describe("Test if drawPlayers() draws players on canvas", () => {
  const canvas = document.createElement("canvas");
  const ctx = canvas.getContext("2d");
  const player1 = new Player("TestPlayer1", 1, "red", "red", "100", "100", Math.PI);
  const player2 = new Player("TestPlayer2", 2, "blue", "blue", 400, 400, Math.PI); 

  let player = [];

  player[0] = player1;
  player[1] = player2;

  const renderer = new Renderer();
  
  test("", () => {
    renderer.drawPlayers(player, 0, canvas, ctx);

    //Saves method calls
    const events = ctx.__getEvents();

    expect(events).toMatchSnapshot();
  })

})


//Test function drawLines()
describe("Test if drawLines() draws lines of players on canvas", () => {
  const canvas = document.createElement("canvas");
  const ctx = canvas.getContext("2d");
  const player1 = new Player("TestPlayer1", 1, "red", "red", "100", "100", Math.PI);
  const player2 = new Player("TestPlayer2", 2, "blue", "blue", 400, 400, Math.PI); 
  player2.isDrawing = false;

  let player = [];

  player[0] = player1;
  player[1] = player2;

  const renderer = new Renderer();
  renderer.borderDrawn = true;

  test("Test drawLines with borderDrawn is true", () => {
    renderer.drawLines(player, 0, canvas, ctx);

    //Saves method calls
    const events = ctx.__getEvents();

    expect(events).toMatchSnapshot();
  })
})

describe("Test if drawLines() draws lines of players on canvas", () => {
  const canvas = document.createElement("canvas");
  const ctx = canvas.getContext("2d");
  const player1 = new Player("TestPlayer1", 1, "red", "red", "100", "100", Math.PI);
  const player2 = new Player("TestPlayer2", 2, "blue", "blue", 400, 400, Math.PI); 
  player2.isDrawing = false;

  let player = [];

  player[0] = player1;
  player[1] = player2;

  const renderer = new Renderer();
  renderer.borderDrawn = false;

  test("Test drawLines with borderDrawn is false", () => {
    renderer.drawLines(player, 0, canvas, ctx);

    //Saves method calls
    const events = ctx.__getEvents();

    expect(events).toMatchSnapshot();
  })
})

describe("Test if drawCountdown() draws the correct countdown on canvas", () => {
  const canvas = document.createElement("canvas");
  const ctx = canvas.getContext("2d");
  const player1 = new Player("TestPlayer1", 1, "red", "red", "100", "100", Math.PI);
  const player2 = new Player("TestPlayer2", 2, "blue", "blue", 400, 400, Math.PI);


  const countDownCanvas = document.createElement("canvas");
  const countDownctx = canvas.getContext("2d");

  let player = [];

  player[0] = player1;
  player[1] = player2;

  const renderer = new Renderer();

  test("Test if the countdown matches the snapshot.", () => {
    renderer.drawPlayers(player, 0, canvas, ctx);
    renderer.drawCountdown(player, 0, countDownCanvas, countDownctx, 1);

    //Saves method calls
    const events = ctx.__getEvents();

    expect(events).toMatchSnapshot();
  })

})


describe("Test if clearCountdown() clears the countdown on canvas", () => {
  const canvas = document.createElement("canvas");
  const ctx = canvas.getContext("2d");
  const player1 = new Player("TestPlayer1", 1, "red", "red", "100", "100", Math.PI);
  const player2 = new Player("TestPlayer2", 2, "blue", "blue", 400, 400, Math.PI);


  const countDownCanvas = document.createElement("canvas");
  const countDownctx = canvas.getContext("2d");

  let player = [];

  player[0] = player1;
  player[1] = player2;

  const renderer = new Renderer();

  test("Test if the countdown was cleared and matches the snapshot.", () => {
    renderer.drawPlayers(player, 0, canvas, ctx);
    renderer.drawCountdown(player, 0, countDownCanvas, countDownctx, 1);
    renderer.clearCountdown(countDownCanvas,countDownctx);

    //Saves method calls
    const events = ctx.__getEvents();

    expect(events).toMatchSnapshot();
  })

})

