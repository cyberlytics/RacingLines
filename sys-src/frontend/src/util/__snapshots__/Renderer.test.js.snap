// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if drawLines() draws lines of players on canvas Test drawLines with borderDrawn is false 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#2196f3",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": 10,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
  Object {
    "props": Object {
      "value": "#ff0000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": 7.699999999999999,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
]
`;

exports[`Test if drawLines() draws lines of players on canvas Test drawLines with borderDrawn is true 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#ff0000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": 7.699999999999999,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "lineWidth",
  },
]
`;

exports[`Test if drawPlayers() draws players on canvas  1`] = `
Array [
  Object {
    "props": Object {
      "value": "#ff0000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": "#ff0000",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {
      "value": "#0000ff",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "strokeStyle",
  },
  Object {
    "props": Object {
      "value": "#0000ff",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
]
`;
